hxd/res/BDFFont.hx

```diff
@@ -0,0 +1,437 @@
+package hxd.res;
+
+#if (haxe_ver < 4)
+import haxe.xml.Fast in Access;
+#else
+import haxe.xml.Access;
+#end
+
+import hxd.BitmapData;
+import haxe.Int32;
+
+/**
+ * Intermediate representation of a glyph. Only used while
+ * parsing a BDF font file.
+ */
+ class BDFFontChar{
+    public var code : Int;
+    public var x : Int;
+    public var y : Int;
+    public var width : Int;
+    public var height : Int;
+    public var xoffset : Int;
+    public var yoffset : Int;
+    public var stride : Int;
+    public var bits : Array<Int>;
+
+    public function new( code, width, height, xoffset, yoffset, stride ) {
+		this.code = code;
+        this.width = width;
+        this.height = height;
+        this.xoffset = xoffset;
+        this.yoffset = yoffset;
+        this.stride = stride;
+        this.bits = new Array();
+    }
+    
+    static public function sortOnHeight( a : BDFFontChar, b : BDFFontChar ){
+        return b.height-a.height; // Largest first
+    }
+}
+
+/**
+ * Parse BDF font format to h2d.Font
+ */
+class BDFFont extends Resource {
+
+    private static inline var BitmapPad : Float = 0.1;
+    private static inline var BitmapMaxWidth : Int = 1024;
+    private static inline var ClearColor : Int32 = 0x000000FF;
+    private static inline var PixelColor : Int32 = 0x00FFFFFF;
+
+    var font : h2d.Font;
+    var bitsPerPixel : Int = 1;
+    var ascent : Int = -1;
+    var descent : Int = -1;
+    var fbbHeight : Int = -1;
+    var glyphData : Array<BDFFontChar>;
+
+	/**
+	 * Convert BDF resource to a h2d.Font instance
+	 * @return h2d.Font The font
+	 */
+    @:access(h2d.Font)
+	public function toFont() : h2d.Font {
+        if ( font != null ) return font;
+        
+		// File starts with STARTFONT
+        if ((entry.getBytes().getInt32(0)!=0x52415453)||(entry.getBytes().getInt32(8)!=0x2E322054))
+            throw "File does not appear to be a BDF file. Should start with STARTFONT";
+
+        // Init empty font
+        font = new h2d.Font(null, 0);
+
+        // Break file into lines
+        var lines = entry.getBytes().toString().split("\n");
+        var linenum = 0;
+
+        // Parse the header
+        linenum = parseFontHeader(lines, linenum);
+        // Parse the glyphs
+        linenum = parseGlyphs(lines, linenum);
+        // Generate glyphs and bitmap
+        generateGlyphs();
+
+        // Return the generated font
+        return font;
+    }
+    
+   /**
+     * Extract what we can from the font header. Unlike other font formats supported by heaps, some
+     * of the values need to be infered from what is given (e.g. line height is not specificed directly,
+     * nor is baseline).
+     * @param lines     The remaining lines in the file
+     * @param linenum   The current line number
+     * @return Int      The final line number after processing header
+     */
+     @:access(h2d.Font)
+     private function parseFontHeader( lines : Array<String>, linenum : Int ) : Int{
```

Comment:
Code style does not use explicit `private function`, just `private`, same goes for `private static inline var`

Reply:
Sorry, I'm not sure what you mean. Do you mean that `private function parseFontHeader(...)` should read `private parseFontHeader(...)`? I have not seen this elsewhere in the source. or do you mean I shouldn't be using the `private` keyword?

Reply:
Yes, if access not specified it defaults to `private`, and Heaps does not use explicit declaration for private variables and methods.